vstruct/cparse.py

```diff
@@ -173,3 +201,43 @@ class awesome(CVStruct):
         int *q;
     };
     '''
+
+# stolen from cod3monk's (https://github.com/cod3monk) PR to pycparser at https://github.com/eliben/pycparser/pull/103/files
+def clean_code(code, comments=True, macros=False):
+    """ Naive comment and macro striping from source code
+        comments:
+            If True, all comments are stripped from code
+        macros:
+            If True, all macros are stripped from code
+        Returns cleaned code. Line numbers are preserved with blank lines,
+        and multiline comments and macros are supported. BUT comments-like
+        strings are (wrongfuly) treated as comments.
+    """
+    if macros:
+        lines = code.split('\n')
```

Comment:
this type of non-context aware parsing of C code is dangerous and error prone. Could we use the the actual AST parser to identify and remove these constructs? I can come up with fairly trivial cases that are broken for this case.

Reply:
unfortunately, pycparser expects its input to be run through actual GCC-e or CC to handle preprocessing beforehand.

Reply:
but your point is well taken.  this is causing unittests to fail.

Reply:
actually, it was the comment clipping code that had a bug.  fixed